index.rst

```diff
@@ -1,61 +1,311 @@
+Note that the process would be similar if we where to use ``LSSTCam`` instead of ``ComCam``, simply replacing the `ComCam`_ class by the equivalent when that becomes available.
+This means, SAL Scripts can be written for ``ComCam`` in a way that will be reusable for the main camera.
+
+.. _MTCS: https://ts-observatory-control.lsst.io/user-guide/maintel/mtcs-user-guide.html
+.. _ComCam: https://ts-observatory-control.lsst.io/user-guide/maintel/CCCamera-user-guide.html
+
+.. _user-provided-optical-aberration:
+
+User-provided optical aberration
+--------------------------------
+
+In some conditions is may be useful to allow users to specify optical aberrations to be introduced to the system, e.g., add a certain amount of comma to the system.
+The MTAOS can easily receive a list of Zernike coefficients, pass them through the Optical Feedback Control and compute the offsets and bending modes required.
+
+The command would simply receive an array of wavefront errors and apply them to the system properly registering the correction internally.
+
+.. code-block:: py
+
+  import numpy as np
+
+  # Assuming the wavefront correction is an array of 19 Zernike coefficients
+  wf = np.zeros(19)
+
+  # Assuming ANSI indexes, wf[7] is vertical comma
+  # Adding 1mm of vertical comma
+  wf[7] = 1.
+
+  await mtaos.cmd_runOFC.set_start(wf = wf)
+
+When the ``MTAOS`` is configured it will load appropriate configuration files for the OFC.
+During commissioning it is likely we will want to run the ``OFC`` with a different set of configurations.
+In order to allow quick turnaround, the command can accept an additional yaml string with overriding  parameters for the currently loaded configuration.
+
+For instance, one could truncate the sensitivity matrix so that a wavefront error correction is applied to a specific component (or components), e.g.;
+
+.. code-block:: py
+
+  import yaml
+
+  # switch off all corrections except m2 hexapod
+  # See https://github.com/lsst-ts/ts_ofc/blob/master/policy/zkAndDofIdxArraySet.yaml
+  ofc_config = {
+    "dofIdx": {
+        "m2Hex": [1, 1, 1, 1, 1],
+        "camHex": [0, 0, 0, 0, 0],
+        "m1m3Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "m2Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      }
+  }
+
+  # convert dictionary to yaml string
+  config = yaml.safe_dump(ofc_config)
+
+  await mtaos.cmd_runOFC.set_start(wf = wf, config = config)
+
+Similarly, the user can manipulate the configuration for Camera hexapod, M2 and M1M3.
+Furthermore, this gives users the capability of customizing the configuration of the OFC on the fly, without the need to recycle the CSC state and the need for several different configurations.
+By relying on a yaml string to parse the configuration it is also possible to expand the feature as needed without the need to update the xml interface, which considerably simplify the redeployment cycle.
```

Comment:
This argument depends on the parameters available to do the overwritten. This might work for a small subset of parameters (and we know what they are). Otherwise, it is easier to maintain the configuration file level in `ts_config_mttcs`. BTW, this is the reason to have this repository to let the users to change different version of configuration file to use. It is easier to just maintain a single interface.